components/script/dom/eventtarget.rs

```diff
@@ -179,22 +179,15 @@ impl CompiledEventListener {
                     }
 
                     CommonEventHandler::BeforeUnloadEventHandler(ref handler) => {
-                        if let Some(event) = event.downcast::<BeforeUnloadEvent>() {
-                            let rv = event.ReturnValue();
-
-                            if let Ok(value) = handler.Call_(object,
-                                                             event.upcast::<Event>(),
-                                                             exception_handle) {
-                                match value {
-                                    Some(value) => {
-                                        if rv.is_empty() {
-                                            event.SetReturnValue(value);
-                                        }
-                                    }
-                                    None => {
-                                        event.upcast::<Event>().PreventDefault();
+                        if let Ok(value) = handler.Call_(object, event, exception_handle) {
```

Comment:
Can you add some tests for `returnValue`? Stick them in `event-handler-processing-algorithm.html` along with the others.